src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -607,10 +531,21 @@ private int FindEndPosition(ReadOnlySpan<char> input, int i)
 
             while (i < input.Length)
             {
-                int j = Math.Min(input.Length, i + AntimirovThresholdLeeway);
-                bool done = _builder._antimirov ?
-                    FindEndPositionDeltas<AntimirovTransition>(input, ref i, j, ref state, ref i_end) :
-                    FindEndPositionDeltas<BrzozowskiTransition>(input, ref i, j, ref state, ref i_end);
+                bool done;
+                if (_builder._antimirov)
+                {
+                    NfaState<TSetType> qNfa = (state is DfaState<TSetType> q1 ? new NfaState<TSetType>(q1._dfaMatchingState, perThreadData) : (NfaState<TSetType>)state);
+                    done = FindEndPositionDeltas(input, ref i, input.Length, ref qNfa, ref i_end, perThreadData);
+                    state = qNfa;
+                }
+                else
+                {
+                    Debug.Assert(state is DfaState<TSetType>);
+                    int j = Math.Min(input.Length, i + AntimirovThresholdLeeway);
+                    var qDfa = (DfaState<TSetType>)state;
+                    done = FindEndPositionDeltas(input, ref i, j, ref qDfa, ref i_end, perThreadData);
+                    state = qDfa;
```

Comment:
Ouch.  Similarly, every time we're storing qDfa or qNfa, which are structs, into state, which is an interface, it's boxing.

The previous code was avoiding the boxing by using a T constrained to the interface, but this method is using `ref ICurrentState<TSetType> state` rather than `ref T state ... where T : ICurrentState<TSetType>`, and as such it's actually a reference type, and if a struct is assigned to it, that struct gets boxed.